### DISCUSSION TITLE
Zoom settings for local screen display not being saved for new based Firefox release

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1429] [Bug Reports]
Now that I have been able to update to 20230918, which replaces Chromium with Firefox for local displays, it looks like the Zoom settings on  Features / Others / Browser on RPi are not being saved.  I normally use a 70% Zoom level fro the Official RPi 7in Display but this doesn't seem to be being saved and the the display reverts back to 100% Zoom level on a reboot.  

Could this be related to the 'Save Setting on a Shared Data system' problem originally reported as #1356 ?  If so, it would seem sensible only to store music library related settings to the Shared Data location with individual system settings (volume levels, display options etc) being stored locally on each system.

Chris

**Comment by Mightyflea64** [comment]
I have the same problem with this release. On startup or after reboot, you have to set the zoom factor again to have a normal screen layout. Hopefully you are able to fix this soon.
Thank you for your hard work :-)

**Reply by feebb** [reply]
I have a related problem with the local screen since the latest release:  I can only scroll vertically on the far right--it appears that there is a narrow scroll bar there. I used to be able to scroll up or down anywhere on the screen.
Could this behavior be related to the new browser?

**Reply by Mightyflea64** [reply]
You are correct, I have the same problem.

**Comment by Mightyflea64** [comment]
I installed update 20231001. Unfortunately this update doesn't solve formentioned problems.
I use the RPi 4B with an official RPi 7" display.

**Comment by ceejayemm** [comment]
@rern 
Updated to 20231001 but no discernible change as far as I can see.  Browser on RPi status box shows:

firefox 116.0.3-1
# cat /srv/http/data/system/localbrowser.conf
rotate=0
zoom=70
screenoff=10
onwhileplay=
hdmi=
cursor=

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; enabled; preset: disabled)
     Active: active (running) since Sun 2023-10-01 17:52:13 BST; 20min ago
   Main PID: 3211 (sudo)
      Tasks: 151 (limit: 1006)
        CPU: 4min 14.987s
     CGroup: /system.slice/localbrowser.service
             ├─3211 /usr/bin/sudo /usr/bin/startx
             ├─3216 /bin/sh /usr/bin/startx
             ├─3231 xinit /etc/X11/xinit/xinitrc -- /etc/X11/xinit/xserverrc :0 -auth /tmp/serverauth.zZxyduAZVt
             ├─3232 /usr/lib/Xorg -nolisten tcp :0 -auth /tmp/serverauth.zZxyduAZVt
             ├─3238 /bin/bash /etc/X11/xinit/xinitrc
             ├─3251 matchbox-window-manager -use_cursor no
             ├─3267 /usr/lib/firefox/firefox -kiosk http://localhost
             ├─3459 /usr/lib/firefox/firefox -contentproc -parentBuildID 20230816205122 -prefsLen 20085 -prefMapSize 231158 -appDir /usr/lib/firefox/browser {a20b3835-3c06-4e6b-8895-6896a911cdb2} 3267 socket
             ├─3505 /usr/lib/firefox/firefox -contentproc -childID 1 -isForBrowser -prefsLen 20481 -prefMapSize 231158 -jsInitLen 242416 -parentBuildID 20230816205122 -appDir /usr/lib/firefox/browser {1b83cdd7-6a15-4aa7-a3dd-0767e74a8c02} 3267 tab
             ├─3525 /usr/lib/firefox/firefox -contentproc -childID 2 -isForBrowser -prefsLen 20149 -prefMapSize 231158 -jsInitLen 242416 -parentBuildID 20230816205122 -appDir /usr/lib/firefox/browser {07d3d342-3094-43ae-8d93-acf2afa516ad} 3267 tab
             ├─3555 /usr/lib/firefox/firefox -contentproc -childID 3 -isForBrowser -prefsLen 20498 -prefMapSize 231158 -jsInitLen 242416 -parentBuildID 20230816205122 -appDir /usr/lib/firefox/browser {75fe1e4c-4472-4d48-b96d-218101428abb} 3267 tab
             └─3582 /usr/lib/firefox/firefox -contentproc -childID 4 -isForBrowser -prefsLen 26599 -prefMapSize 231158 -jsInitLen 242416 -parentBuildID 20230816205122 -appDir /usr/lib/firefox/browser {19a6eb65-41cf-4397-9a0d-c08e6a08f66e} 3267 tab

Oct 01 17:52:14 RAConservatory sudo[3232]:         (++) from command line, (!!) notice, (II) informational,
Oct 01 17:52:14 RAConservatory sudo[3232]:         (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
Oct 01 17:52:14 RAConservatory sudo[3232]: (==) Log file: "/var/log/Xorg.0.log", Time: Sun Oct  1 17:52:14 2023
Oct 01 17:52:14 RAConservatory sudo[3232]: (==) Using config directory: "/etc/X11/xorg.conf.d"
Oct 01 17:52:14 RAConservatory sudo[3232]: (==) Using system config directory "/usr/share/X11/xorg.conf.d"
Oct 01 17:52:16 RAConservatory sudo[3240]: Failed to connect to bus: No medium found
Oct 01 17:52:16 RAConservatory sudo[3241]: dbus-update-activation-environment: error: unable to connect to D-Bus: Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead
Oct 01 17:52:16 RAConservatory sudo[3251]: matchbox-window-manager: Failed to connect to session manager
Oct 01 17:52:17 RAConservatory sudo[3264]: grep: /root/.mozilla/firefox/profiles.ini: No such file or directory

There also errors from journalctl -b but I am not sure is they are related, see lines 1208 to 1327 (amongst others) in the uploaded journalctl eport file (journal.txt)

Chris

[journal.txt](https://github.com/rern/rAudio/files/12777780/journal.txt)

**Comment by TonG60** [comment]
I use a 3.5” touch screen display on a Raspberry Pi 3. With rAudio release 20230918 Zoom Out on the local Firefox is reset to 100 % at every reboot. At this moment the problem persists in the new release 202310131. Without the correct zoom out the menu bars on top and bottom on the small display are missing.

Simple solution is to connect a USB keyboard and press the combination CRTL- once or more. In my case three times make the menu bars and all menu items of the cogwheel in de upper right corner visible.

Another way is to press Alt F4, this will close the local web browser. Wait a little while so Firefox can close fully, it's slow. Next under Features turn Browser on Rpi on, it should be turned off now. The local browser will start, but be patient, this also takes a while. The zoom setting chosen in this menu will be used now. But, at reboot zoom factor is 100 % again.

Proposed temporary (?) solution to simulate keys pressed at boot:

Install the xorg-xhost package and xdotool package

In /usr/bin add a script called zoomout.sh containing the following lines:

#!/bin/bash

sleep 60
XAUTHORITY=/home/root/.Xauthority
sudo xhost +
DISPLAY=:0 /usr/bin/xdotool key ctrl+minus
sleep 1
DISPLAY=:0 /usr/bin/xdotool key ctrl+minus
sleep 1
DISPLAY=:0 /usr/bin/xdotool key ctrl+minus



To start the zoomout.sh script at boot create a service file in /etc/systemd/system/ called zoomout.service containing following lines:

[Unit]
Description=Zoom out in local Firefox web browser

[Service]
Environment="DISPLAY=:0"
Type=simple
ExecStart=/usr/bin/zoomout.sh

[Install]
WantedBy=multi-user.target


This service must be enabled. Before enabling your service, change the permissions to 644:

	sudo chmod 644 /etc/systemd/system/zoomout.service

After creating this file or every time the file is changed the daemon must be reloaded:

	sudo systemctl daemon-reload

Enable the service so it will run at startup:

	sudo systemctl enable zoomout


To verify the service is working, start the service manually and check its status:

	sudo systemctl start zoomout

	systemctl status zoomout

It took me many many hours to come up with this solution. I’m not that familiar with creating scripts and so maybe a line or more (leftovers from previous attempts) is not necessary(?). I’m a user not a programmer. If anyone can perfect it or has a simpler solution, please share. I hope I haven’t forgotten to mention anything. The above solution works for me. In zoomout.sh I use sleep 60 to wait 60 seconds so Firefox has enough time to start. In my case two extra zoomouts are needed, each with a 1 second delay for safety, start about 5 to 7 seconds after the local browser content appears. If rern has a solution, in a next release, comment the xdotool lines in /usr/bin/zoomout.sh. The script will do nothing then. I really like rAudio and think rern does a great job!

**Comment by R101G** [comment]
I am playing with a 7" 800x480 touch screen on a Pi4B and find the brightness setting is not saved. Possibly a related problem?

Also the bottom bar height is curtailed at the lower edge, which might not be a related issue.

**Comment by ceejayemm** [comment]
@rern 
Seems fixed for me with 20231020.

Thanks

Chris
